packages/api/lib/rulesHelpers.js

```diff
@@ -273,25 +274,25 @@ async function deleteRuleResources(knex, rule) {
  *
  * Avoids object mutation by cloning the original rule item.
  *
- * @param {PostgresRule} ruleItem - A rule item
+ * @param {ApiRule} ruleItem - A rule item
  * @param {string} snsSubscriptionArn
  *   UUID for event source mapping from SNS topic to messageConsumer Lambda
- * @returns {PostgresRule} - Updated rule item
+ * @returns {ApiRule} - Updated rule item
  */
 function updateSnsRuleArn(ruleItem, snsSubscriptionArn) {
   const updatedRuleItem = cloneDeep(ruleItem);
-  updatedRuleItem.arn = snsSubscriptionArn;
+  updatedRuleItem.rule.arn = snsSubscriptionArn;
```

Comment:
I was wondering if we should revert this back to the previous version of the logic since we're handling an API rule now, but if the unit tests are passing maybe we don't need to worry about it

Reply:
Nope we shouldn't